UnityProject/Assets/Scripts/Objects/Research/BlastYieldDetector.cs

```diff
@@ -0,0 +1,325 @@
+			if (awarded> 0)
+			{
+				Chat.AddLocalMsgToChat($"Research points awarded: {awarded.ToString()}. New Total for Ordnance is {points.ToString()}", gameObject);
+			}
+			else
+			{
+				Chat.AddLocalMsgToChat("Explosion strength not close enough to yield " +
+				                       "target to award additional research.",gameObject);
+			}
+
+			return awarded;
+		}
+
+		/// <summary>
+		/// Determines the research point value of an explosion based on the following formula.
+		/// [ max(e^(-(x - a1)^2/1250000)×140, e^(-(x - a2)^2/5000000)×70) ]
+		/// a1 and a2 are the per round randomised values set between 1000 and 19000, modelled after minimum and maximum
+		/// values for ExplosionBase ExplosionStrengths, that are targets to reach for blast yield.
+		/// </summary>
+		/// <param name="explosiveStrength"></param>
+		/// <returns></returns>
+		private int calculateResearchPoints(float explosiveStrength)
+		{
+			float term1 = (float)Math.Exp(-Math.Pow(explosiveStrength - maxPointYieldTarget, 2) / 1250000) *
+			              maxPointsValue;
+			float term2 = (float)Math.Exp(-Math.Pow(explosiveStrength - easyPointYieldTarget, 2) / 5000000) *
+			              easyPointsValue;
+			return (int)Mathf.Max(term1, term2);
+		}
+
+		#region Multitool Interaction Overrides
+
+		public override void SubscribeToServerEvent(ResearchServer server)
+		{
+			base.SubscribeToServerEvent(server);
+			ExplosiveBase.ExplosionEvent.AddListener(DetectBlast);
+			Chat.AddLocalMsgToChat("Server connection found: Monitoring.",gameObject);
+			stateSync = BlastYieldDetectorState.Connected;
+			GetYieldTargets();
+			serverConnEvent(true);
+		}
+
+		public override void UnSubscribeFromServerEvent()
+		{
+			base.UnSubscribeFromServerEvent();
+			ExplosiveBase.ExplosionEvent.RemoveListener(DetectBlast);
+
+			if (PoweredState != PowerState.Off)
+			{
+				Chat.AddLocalMsgToChat("Lost server connection.", gameObject);
+				stateSync = BlastYieldDetectorState.Broken;
+			}
+
+			blastData.Clear();
+			easyPointYieldTarget = 0;
+			maxPointYieldTarget = 0;
+			serverConnEvent(false);
+		}
+
+		#endregion Multitool Interaction Overrides
+
+		#region ICanOpenNetTab
+
+		public bool CanOpenNetTab(GameObject playerObject, NetTabType netTabType)
+		{
+			if (PoweredState == PowerState.Off)
+			{
+				Chat.AddExamineMsgFromServer(playerObject, $"{gameObject.ExpensiveName()} is unpowered");
+				return false;
+			}
+			return true;
+		}
+
+		#endregion ICanOpenNetTab
+
+		#region Interactions
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Wrench))
+			{
+				return interaction.Intent != Intent.Harm;
+			}
+			return false;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (stateSync == BlastYieldDetectorState.Connected)
+			{
+				if (Validations.HasUsedItemTrait(interaction, CommonTraits.Instance.Wrench))
```

Comment:
Can you invert these two conditions to reduce nesting?

Reply:
done